core/zoe/docs/zoe.md

```diff
@@ -0,0 +1,263 @@
+# Zoe: Offer-Safety Enforcement
+
+Note: Zoe is currently at the pre-alpha stage. It has not yet been
+formally tested or hardened.
+
+This guide assumes some knowledge of the [ERTP fundamentals](./README.md).
+
+## What is Zoe? 
+
+__For users__: Zoe guarantees that as a user of a smart contract, you
+will either get what you wanted or get a refund back, even if the
+smart contract is buggy or backdoored. (In fact, the smart contract
+never has access to your digital assets.)
+
+__For developers__: Zoe provides a safety net so you can focus on what
+your smart contract does best, without worrying about your users
+losing their assets due to a bug in the code that you wrote. Writing a
+smart contract on Zoe is easy: all of the Zoe smart contracts are
+written in the familiar language of JavaScript. 
+
+## Sounds like magic. How does it actually work?
+
+To use Zoe, we have to put things in terms of "offers". An offer is a
+statement about what you want and what you're willing to offer. It
+turns out, many smart contracts (apart from gifts and one-way
+payments) involve an exchange of digital assets that can be put in
+terms of offers. 
+
+In this version of Zoe, our offers are simple (see [our roadmap](TODO) for
+more complex offer types and their expected release). We can say
+things like, "I'll give you [three
+wood for two bricks](https://en.wikipedia.org/wiki/Catan)." We can
+also say something like, "I'll give you three wood, and *the most* I'm
+willing to pay is two bricks." Or even: "I can pay you two bricks and
+I expect *at least* three wood back." [Learn more about the particulars
+of structuring an offer here](TODO). 
+
+Offers are a structured way of describing user intent. To a certain
+extent, an offer description is the user's *contractual understanding*
+of the agreement they are entering into. You might have noticed that
+the offer doesn't specify the mechanism by which the exchange happens.
+The offer doesn't say whether the item you want is up for auction, in
+an exchange, or part of a private trade. The offer doesn't mention the
+particular mechanism because an important part of the design of Zoe is
+a __separation of concerns__. Zoe is responsible for enforcing what we
+call "offer-safety", and the smart contract that runs on top of Zoe is
+responsible for figuring out a proposed reallocation of resources. To
+use an auction as an example, the smart contract is responsible for
+figuring out who wins the auction and how much they pay, but Zoe
+handles the escrowing of the bids and the payments. You can think of
+this as similar to e-commerce websites using a separate
+payment-processor so that they don't have to handle the credit cards
+themselves.
+
+### What is "offer-safety"?
+
+Zoe guarantees offer-safety, meaning that when a user makes an offer
+that is escrowed with Zoe, Zoe guarantees that the user will either
+get what they said they wanted, or the user will get what they
```

Comment:
"get what the originally offered back" --> "get back what they originally offered".

Reads better. Needs less mental stack to parse.